File: jdroid-gradle-googleplay-publisher-plugin/src/main/java/com/jdroid/android/googleplay/publisher/task/BaseTask.java
package com.jdroid.android.googleplay.publisher.task;

import com.jdroid.android.googleplay.publisher.App;
import com.jdroid.android.googleplay.publisher.AppContext;
import com.jdroid.android.googleplay.publisher.GooglePlayPublisherPluginExtension;
import com.jdroid.android.googleplay.publisher.TrackType;
import com.jdroid.android.googleplay.publisher.commons.AbstractTask;
import com.jdroid.java.utils.StringUtils;

public abstract class BaseTask extends AbstractTask {

	@Override
	protected void onExecute() {
		
		GooglePlayPublisherPluginExtension extension = getExtension();
		
		AppContext appContext = new AppContext();
		appContext.setApplicationId(extension.getApplicationId());
		appContext.setPrivateKeyJsonFilePath(extension.getPrivateKeyJsonFilePath());
		appContext.setLocales(StringUtils.splitWithCommaSeparator(extension.getLocales()));
		appContext.setReleaseName(extension.getReleaseName());
		appContext.setDraft(extension.getDraft());
		
		appContext.setDryRun(extension.getDryRun());

		appContext.setMetadataPath(extension.getMetadataPath());

		appContext.setReleaseNotesRequired(extension.getReleaseNotesRequired());
		appContext.setVideoRequired(extension.getVideoRequired());
		appContext.setPromoGraphicRequired(extension.getPromoGraphicRequired());
		appContext.setPhoneScreenshotsRequired(extension.getPhoneScreenshotsRequired());
		appContext.setTvBannerRequired(extension.getTvBannerRequired());
		appContext.setSevenInchScreenshotsRequired(extension.getSevenInchScreenshotsRequired());
		appContext.setTenInchScreenshotsRequired(extension.getTenInchScreenshotsRequired());
		appContext.setTvScreenshotsRequired(extension.getTvScreenshotsRequired());
		appContext.setWearScreenshotsRequired(extension.getWearScreenshotsRequired());

		appContext.setApkPath(extension.getApkPath());
		appContext.setApkDir(extension.getApkDir());
		appContext.setBundlePath(extension.getBundlePath());
		appContext.setBundleDir(extension.getBundleDir());
		
		appContext.setTrackType(TrackType.findByKey(extension.getTrack()));
		appContext.setUserPercentage(extension.getUserPercentage());

		appContext.setFailOnApkUpgradeVersionConflict(extension.getFailOnApkUpgradeVersionConflict());

		onExecute(new App(appContext));
	}

	protected abstract void onExecute(App app);

}

File: jdroid-gradle-googleplay-publisher-plugin/src/main/java/com/jdroid/android/googleplay/publisher/task/BaseTask.kt
package com.jdroid.android.googleplay.publisher.task

import com.jdroid.android.googleplay.publisher.App
import com.jdroid.android.googleplay.publisher.AppContext
import com.jdroid.android.googleplay.publisher.TrackType
import com.jdroid.android.googleplay.publisher.commons.AbstractTask
import com.jdroid.java.utils.StringUtils

abstract class BaseTask : AbstractTask() {

    override fun onExecute() {
        val extension = getExtension()
        val appContext = AppContext()
        appContext.applicationId = extension.applicationId
        appContext.privateKeyJsonFilePath = extension.privateKeyJsonFilePath
        appContext.connectTimeout = extension.connectTimeout
        appContext.readTimeout = extension.readTimeout
        appContext.locales = StringUtils.splitWithCommaSeparator(extension.locales)
        appContext.releaseName = extension.releaseName
        appContext.isDraft = extension.draft
        appContext.isDryRun = extension.dryRun
        appContext.metadataPath = extension.metadataPath
        appContext.deobfuscationFilePath = extension.deobfuscationFilePath
        appContext.isDeobfuscationFileUploadEnabled = extension.isDeobfuscationFileUploadEnabled
        appContext.isReleaseNotesRequired = extension.releaseNotesRequired
        appContext.isVideoRequired = extension.videoRequired
        appContext.isPromoGraphicRequired = extension.promoGraphicRequired
        appContext.isPhoneScreenshotsRequired = extension.phoneScreenshotsRequired
        appContext.isTvBannerRequired = extension.tvBannerRequired
        appContext.isSevenInchScreenshotsRequired = extension.sevenInchScreenshotsRequired
        appContext.isTenInchScreenshotsRequired = extension.tenInchScreenshotsRequired
        appContext.isTvScreenshotsRequired = extension.tvScreenshotsRequired
        appContext.isWearScreenshotsRequired = extension.wearScreenshotsRequired
        appContext.apkPath = extension.apkPath
        appContext.apkDir = extension.apkDir
        appContext.bundlePath = extension.bundlePath
        appContext.bundleDir = extension.bundleDir
        appContext.trackType = TrackType.findByKey(extension.track)
        appContext.userPercentage = extension.userPercentage
        appContext.setFailOnApkUpgradeVersionConflict(extension.failOnApkUpgradeVersionConflict)
        onExecute(App(appContext))
    }

    protected abstract fun onExecute(app: App?)
}
